### PR TITLE
DVC-8633: feat: add overrides update

### PR DESCRIPTION
**Pull request by suthar26**
- [x] happy path without flags
- [x] need to test headless mode and without some flags
- [x] need to test when user id is not set
- [x] improve the output to be user relevant info only

### CODE REVIEW

**Comment by suthar26** [comment]
Current dependencies on/for this PR:
* main
  * **PR #277** <a href="https://app.graphite.dev/github/pr/DevCycleHQ/cli/277" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/DevCycleHQ/cli/277?utm_source=stack-comment).

File: README.md [hunk 1]
```diff
@@ -1,5 +1,4 @@
-DevCycle CLI
-=================
+# DevCycle CLI
```
Comment (ajwootto) [inline README.md:1]: how come this generated differently?

File: src/api/zodClient.ts [hunk 2]
```diff
@@ -756,6 +756,20 @@ const ProjectUserProfile = z.object({
 const UpdateProjectUserProfileDto = z.object({
     dvcUserId: z.string().nullable(),
 })
+const Override = z.object({
+    _project: z.string(),
+    _environment: z.string(),
+    _feature: z.string(),
+    _variation: z.string(),
+    dvcUserId: z.string(),
```
Comment (kaushalkapasi) [inline src/api/zodClient.ts:764]: does this support empty strings, null and undefined?

File: README.md (hunk 1 above)
Comment (suthar26) [reply README.md]: dont know but reverted the changes

File: src/api/zodClient.ts (hunk 2 above)
Comment (suthar26) [reply src/api/zodClient.ts:764]: not sure, but we would only get the response for the UpdateCommand if all values are set and this is the return object

File: README.md (hunk 1 above)
Comment (suthar26) [reply README.md]: was a version mismatch, removed node_modules and rebuild and all is good now

File: src/api/zodClient.ts [hunk 3]
```diff
@@ -3310,6 +3326,49 @@ const endpoints = makeApi([
...
+            {
+                status: 403,
+                schema: z.void(),
+            },
+            {
+                status: 404,
+                schema: NotFoundErrorResponse,
+            },
+            {
+                status: 409,
```
Comment (kaushalkapasi) [inline src/api/zodClient.ts:3366]: is a 409 conflict possible when updating an override?

File: src/commands/overrides/update.ts [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+            variation = response.variation._id
+        }
+
+        const params = await this.populateParametersWithZod(UpdateOverrideDto, this.prompts, {
+            feature, environment, variation
+        })
+
+        const result = await updateOverride(this.authToken, this.projectKey, feature, params)
+        this.writer.successMessage(
+            `Successfully added override for userId ${result.dvcUserId}`)
```
Comment (kaushalkapasi) [inline src/commands/overrides/update.ts:91]: can we make the output clearer as to what change was made?

```suggestion
            `Successfully added override for User ID: ${result.dvcUserId} for feature: ${featureKey} on environment: ${environmentKey} to variation: ${variationKey}`)
```

File: src/api/zodClient.ts (hunk 3 above)
Comment (suthar26) [reply src/api/zodClient.ts:3366]: it does not.. removed it

File: src/commands/overrides/update.ts (hunk 4 above)
Comment (suthar26) [reply src/commands/overrides/update.ts:91]: I had that before but it was too long so removed it, but I will prettify it

Comment (kaushalkapasi) [reply src/commands/overrides/update.ts:91]: you can multi-line split by doing something like:
`Successfully added override for User ID: ${result.dvcUserId}` +
` for feature: ${featureKey} ` etc etc

File: src/commands/overrides/update.ts [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+    static flags = {
+        ...Base.flags,
+        feature: Flags.string({
+            description: 'The feature to update an override for'
+        }),
+        environment: Flags.string({
+            description: 'The environment to update an override for'
+        }),
+        variation: Flags.string({
+            description: 'The variation to update an override for'
```
Comment (kaushalkapasi) [inline src/commands/overrides/update.ts:30]: description should be `The variation that will be used as the Override`

File: src/commands/overrides/update.ts [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+import { variationPrompt } from '../../ui/prompts/variationPrompts'
+import { UpdateOverrideDto } from '../../api/schemas'
+import { updateOverride } from '../../api/overrides'
+import Base from '../base'
+import { fetchUserProfile } from '../../api/userProfile'
+
+export default class UpdateOverride extends Base {
+    static hidden = false
+    authRequired = true
+    static description = 'Update an override'
```
Comment (chantalchung) [inline src/commands/overrides/update.ts:17]: To be consistent with the other descriptions and usages, let's capitalize `Override` here, in the flag descriptions, and the error/success/info messages

File: docs/overrides.md [hunk 7]
```diff
@@ -34,3 +35,36 @@ GLOBAL FLAGS
 DESCRIPTION
   Clear Overrides for a given Feature or Project.
 ```
+
+## `dvc overrides update`
+
+Update an override
```
Comment (chantalchung) [inline docs/overrides.md:41]: did you run `yarn build`? how come this didn't update with your changes? 🤔

Comment (suthar26) [reply docs/overrides.md:41]: i did not..

**Review by chantalchung — APPROVED** [review]

**Pull request merged**